[MINOR][PYTHON][DOCS] Fix pyspark.sql.functions.reduce docstring typo

kaashif opened the pull request:
### What changes were proposed in this pull request?

This PR fixes a mistake in the docstring for `pyspark.sql.functions.reduce`. The parameter to the function is called `initialValue` not `zero` - there is no other mention of `zero` on the page, so it must be a mistake and should be `initialValue`.

### Why are the changes needed?

The docstring is incorrect.

### Does this PR introduce _any_ user-facing change?

No


### How was this patch tested?

N/A


### Was this patch authored or co-authored using generative AI tooling?

No

zhengruifeng commented:
thanks for this fix. would you mind enabling the github action? 
you may refer to the `Pull request` section in https://spark.apache.org/contributing.html

allisonwang-db reviewed (approved):
Good find

kaashif committed:
Fix pyspark.sql.functions.reduce docstring typo

Changes `zero` to `initialValue` - `zero` was previously renamed to `initialValue` but the docstring wasn't changed.

zhengruifeng commented:
thanks, merged to master
